go/vumitools/middleware.py

```diff
@@ -603,6 +603,67 @@ def handle_failure(self, failure, connector_name):
         return failure
 
 
+class ConversationMetricsMiddlewareConfig(BaseMiddleware.CONFIG_CLASS):
+
+    redis_manager = ConfigDict(
+        "Redis configuration parameters", default={}, static=True)
+
+    riak_manager = ConfigRiak(
+        "Riak configuration parameters. Must contain at least a bucket_prefix"
+        " key", required=True, static=True)
+
+
+class ConversationMetricsMiddleware(BaseMiddleware):
+    """
+    Middleware that stores which conversations have received or sent messages
+
+    :param dict redis_manager:
+        Connection configuration details for Redis.
+    :param dict riak_manager:
+        Configuration details for Riak.
+    """
+
+    CONFIG_CLASS = ConversationMetricsMiddlewareConfig
+
+    @inlineCallbacks
+    def setup_middleware(self):
+        # We don't use a VumiApi here because we don't have a Riak config for
+        # it.
+        self.vumi_api = yield VumiApi.from_config_async({
+            "riak_manager": self.config.riak_manager,
+            "redis_manager": self.config.redis_manager,
+        })
```

Comment:
We should probably actually avoid using VumiApi here so that we don't need to set up a RiakManager:
* You can get a base redis object with `redis = yield TxRedisManager.from_config(self.config.redis_manager)`.
* Call `redis.close_manager()` in `teardown_middleware`.
* When constructing the `mdh`, pass `None` for the `vumi_api` (since `.get_conversation_key` and `.get_account_key` don't need a `vumi_api`.